endpoint/endpointComp/endpoint.c

```diff
@@ -126,21 +126,24 @@ COMPONENT_INIT {
   srand(time(NULL));
 
   device_t* device = ta_device(STRINGIZE(EP_TARGET));
-  device->op->get_key(private_key);
-  device->op->get_device_id(device_id);
-
+  if (device == NULL) {
+    LE_ERROR("Can not get specific device");
+  } else {
+    device->op->get_key(private_key);
+    device->op->get_device_id(device_id);
 #ifdef ENABLE_ENDPOINT_TEST
-  LE_TEST_INIT;
-  LE_TEST_INFO("=== ENDPOINT TEST BEGIN ===");
-  LE_TEST(SC_OK == send_transaction_information(host, port, ssl_seed, value, message, message_fmt, tag, address,
-                                                next_address, private_key, device_id_ptr, iv));
-  LE_TEST_EXIT;
+    LE_TEST_INIT;
+    LE_TEST_INFO("=== ENDPOINT TEST BEGIN ===");
+    LE_TEST(SC_OK == send_transaction_information(host, port, ssl_seed, value, message, message_fmt, tag, address,
+                                                  next_address, private_key, device_id_ptr, iv));
+    LE_TEST_EXIT;
 #else
-  while (true) {
-    status_t ret = send_transaction_information(host, port, ssl_seed, value, message, message_fmt, tag, address,
-                                                next_address, private_key, device_id_ptr, iv);
-    LE_INFO("Send transaction information return: %d", ret);
-    sleep(10);
-  }
+    while (true) {
+      status_t ret = send_transaction_information(host, port, ssl_seed, value, message, message_fmt, tag, address,
```

Comment:
Do we need to add `TODO listen input from UART here` between line 141 and 142? Since in the real case endpoint should listen the input data from peripheral devices.